src/extension/background-script/CryptoServices/utils.ts

```diff
@@ -28,20 +31,43 @@ export interface TypedMessageText extends TypedMessageMetadata {
     readonly type: 'text'
     readonly content: string
 }
+export interface TypedMessageImage extends TypedMessageMetadata {
+    readonly type: 'image'
+    readonly content: string
+}
 export interface TypedMessageComplex extends TypedMessageMetadata {
     readonly type: 'complex'
     readonly items: readonly TypedMessage[]
 }
 export interface TypedMessageUnknown extends TypedMessageMetadata {
     readonly type: 'unknown'
 }
-export type TypedMessage = TypedMessageText | TypedMessageComplex | TypedMessageUnknown
+export type TypedMessage = TypedMessageText | TypedMessageImage | TypedMessageComplex | TypedMessageUnknown
 export function makeTypedMessage(text: string, meta?: ReadonlyMap<string, unknown>): TypedMessageText
-export function makeTypedMessage(content: string, meta?: ReadonlyMap<string, unknown>): TypedMessage {
+export function makeTypedMessage(buf: ArrayBuffer, meta?: ReadonlyMap<string, unknown>): TypedMessageImage
```

Comment:
I suggest to add another method for it. Binary data doesn't has to be an image. Maybe `makeTypedMessageImage`

Reply:
Have been added a `type` filed for distinguishing different message type in https://github.com/DimensionDev/Maskbook/tree/feature/encrypt-image.

Reply:
I prefer to split into different methods. No good reason to handle this complex function overload